### ISSUE TITLE
[Feature Request] Sum total of rows

### ISSUE THREAD

**Issue by LeCuay** [EddieDover/fvtt-party-sheet#40]
# Summary

Yet another request 😅.

For total party wealth basically but I think it's a great addition in general.

This could get tricky since it'd require for a single row to iterate all actors in order to retrieve values as array or for adding them to a common property.

It might be a good idea to wait for #38

**Comment by EddieDover** [comment]
I like how these feature requests just get more and more complicated.

Honestly not sure how I would pull this off in a graceful manner. I'll have to think about how possible it is and get back to you.

**Comment by EddieDover** [comment]
It took a bit of work but...
![image](https://github.com/user-attachments/assets/1e0bba2e-7583-4147-bfe9-ca8808d88bad)

Now, that being said, I'm not 100% sure I'm happy with this implementation or the jank I had to pull to get this to work. I want nothing more than to give the users of my modules exactly what they want, but I'm not sure the cost of this feature is worth the value it brings.

As it stands, I added a `type` for `total`. Whatever text is fed into total, gets treated like the type is `direct`, however in the background the total is ... totaled and stored elsewhere. During runtime the direct value is displayed, until the bottom of the table where I'm running another loop, similar to how I generate the column headers, that instead generates column totals.  This requires added a special field, just the `text` that was used in the column, so it can be referenced after it's converted by the `direct` parser.

Unfortunately, I'm not sure what I can do about all the extra border lines without adding a ton of customization options and then having to work those in etc.

AND, this really only works for ints. If I make things floats, then I have to worry about precision because if people start using ints, they'll get .00 decimal spots, etc.

I almost want to make a separate section on the popup, with it's own JSON, just to display 'total values' then this might be a lot more clean.

**Comment by LeCuay** [comment]
> It took a bit of work but... ![image](https://private-user-images.githubusercontent.com/464286/359289599-1e0bba2e-7583-4147-bfe9-ca8808d88bad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMjkzNzksIm5iZiI6MTcyNDEyOTA3OSwicGF0aCI6Ii80NjQyODYvMzU5Mjg5NTk5LTFlMGJiYTJlLTc1ODMtNDE0Ny1iZmU5LWNhODgwOGQ4OGJhZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMFQwNDQ0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YjM5ZGZmNDkyY2U1YmJlYTk3NGUyNjNlYTk3ZjhmMTE1NTYxMzM3MWE0MTEyZDJkMjRkOTg4NGRkZDY5M2E3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wA9rhy5MURE63SeE8T2JyZKGlzHHvMdUL1E2uJHCtnY)
> 
> Now, that being said, I'm not 100% sure I'm happy with this implementation or the jank I had to pull to get this to work. I want nothing more than to give the users of my modules exactly what they want, but I'm not sure the cost of this feature is worth the value it brings.
> 
> As it stands, I added a `type` for `total`. Whatever text is fed into total, gets treated like the type is `direct`, however in the background the total is ... totaled and stored elsewhere. During runtime the direct value is displayed, until the bottom of the table where I'm running another loop, similar to how I generate the column headers, that instead generates column totals. This requires added a special field, just the `text` that was used in the column, so it can be referenced after it's converted by the `direct` parser.
> 
> Unfortunately, I'm not sure what I can do about all the extra border lines without adding a ton of customization options and then having to work those in etc.
> 
> AND, this really only works for ints. If I make things floats, then I have to worry about precision because if people start using ints, they'll get .00 decimal spots, etc.
> 
> I almost want to make a separate section on the popup, with it's own JSON, just to display 'total values' then this might be a lot more clean.

Yeah... This one is gonna the most difficult. Not only because of the iteration within the loop but also the information displayed. I recommended to wait for #38 because the solution might go via a single column with full rowspan in order to display party gold

**Comment by EddieDover** [comment]
I actually did this work branching from #38 

And now that I think about it, I did implement colSpan the same way as rowSpan, so there may be a way track the empty cells then colspan between them and use right-aligned text. Not sure.
